### ISSUE TITLE
'npm' is not a meteor command (1.6.0.1)

### ISSUE THREAD

**Issue by wcgcoder** [meteor/meteor#9535]
Hello,

Thanks in advance for the attention to this issue.
It seems that every other build of meteor has this problem since version 1.4. Why does this happen so often?

- The version of Meteor showing the problem.
1.6.0.1

- The last version of Meteor where the problem did _not_ occur, if applicable.
1.6.0

- The operating system you're running Meteor on.
Debian:jessie

- The expected behavior.
meteor runs the npm command as directed

- The actual behavior.
'npm' is not a Meteor command. See 'meteor --help'.

- A **simple** reproduction!

```
RUN set -ex 	&& curl https://install.meteor.com?release=$METEOR_VERSION | /bin/sh
 ---> Running in d8176ab991c1
+ curl https://install.meteor.com?release=1.6.0.1
+ /bin/sh
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  7786    0  7786    0     0  16920      0 --:--:-- --:--:-- --:--:-- 16962
Downloading Meteor distribution

Meteor 1.6.0.1 has been installed in your home directory (~/.meteor).
Writing a launcher script to /usr/local/bin/meteor for your convenience.

To get started fast:

  $ meteor create ~/my_cool_app
  $ cd ~/my_cool_app
  $ meteor

Or see the docs at:

  docs.meteor.com

Removing intermediate container d8176ab991c1
 ---> 2a3ec5ac14aa
Step 14/19 : ENV METEOR_PROFILE=1000000 	METEOR_DEBUG_BUILD=1
 ---> Running in abbc2b37b85f
Removing intermediate container abbc2b37b85f
 ---> 272862ed1b99
Step 15/19 : RUN set -ex 	&& meteor --allow-superuser npm install 
 ---> Running in a125bb716a23
+ meteor --allow-superuser npm install

Even with METEOR_ALLOW_SUPERUSER or --allow-superuser, permissions in your app
directory will be incorrect if you ever attempt to perform any Meteor tasks as
a normal user. If you need to fix your permissions, run the following command
from the root of your project:

  sudo chown -Rh <username> .meteor/local

'npm' is not a Meteor command. See 'meteor --help'.
```

**Comment by benjamn** [comment]
The issue here is that `--allow-superuser` comes before `npm`. Everything should work if you put any flags after the `meteor npm ...` prefix; for example,
```sh
meteor npm --allow-superuser install
```
For performance reasons, `meteor npm ...` and `meteor node ...` commands are handled as a special case, separate from Meteor's usual command parsing system. Specifically, when the first token after `meteor` is `npm`, we essentially just strip `meteor` and run the remaining command as an `npm` command in a child process (after setting up the appropriate environment).

The code responsible for this logic is relatively simple: it doesn't springboard between Meteor releases, it doesn't use Babel, it doesn't do any argument parsing of its own, and it avoids loading the rest of the command-line tool codebase.

If we waited for Meteor's typical argument-parsing code to realize that the command name was `npm`, and only then reinterpreted the command as an `npm` command, we would already have loaded more code than necessary, and every `meteor npm ...` command would take noticeably longer.

In short, please think of `meteor npm ...` as an atomic command prefix, as if it was called `meteor-npm`. Although we could potentially rearrange flags to make this work, it's hard to devote much time to that, given the ease of the workaround.

I hope understanding why this works the way it does is helpful to you!

**Comment by wcgcoder** [comment]
@benjamn Thanks for the response!
That definitely makes sense, and is totally fine with me. 

I've changed my command to place flags after the initial command as you suggested.

**Comment by fyngyrz** [comment]
@wcgcoder Trying to get a dev environment set up for rocketchat.

this...
`meteor npm start`
 ...fails immediately because there is already something running on port 3000

The advice is to provide meteor with `--port XXXX`, so I am using `--port 4000`, which is an okay port for a non-super-user under linux, and one that is not otherwise in use.

I have moved the `--port 4000` to every position on the line except first, and it fails in every one of them:

`meteor --port 4000 npm start` = 'npm' is not a Meteor command. (etc.)
`meteor npm --port 4000 start` = Usage: npm <command> (etc.)
`meteor npm start --port 4000` = '4000' is not a Meteor command. (etc.)

However...
`meteor --port 4000`
..._does_ start things up, although they don't actually work (and to be fair, that could be rocketchat, not meteor or npm or whatever in dog's name is actually going on here.)

Any suggestions?

**Comment by klaussner** [comment]
Try `meteor npm start -- --port 4000`. Arguments after the `--` are forwarded to the npm script—in this case, the `meteor` command. https://docs.npmjs.com/cli/start#synopsis

**Comment by fyngyrz** [comment]
Thank you! It took right off. Much appreciated. :)
